pkg/operator/ceph/disruption/clusterdisruption/osd.go

```diff
@@ -42,25 +40,83 @@ import (
 
 const (
 	// PDBAppName is that app label value for pdbs targeting osds
-	PDBAppName         = "rook-ceph-osd-pdb"
-	disabledPDBKey     = "disabled-pdb"
-	disabledPDBTimeKey = "pdb-disabled-at"
+	PDBAppName            = "rook-ceph-osd-pdb"
+	drainingFailureDomain = "draining-failureDomain"
+	drainingFDTimeKey     = "failureDomain-draining-since"
 	// DefaultMaintenanceTimeout is the period for which a drained failure domain will remain in noout
 	DefaultMaintenanceTimeout = 30 * time.Minute
 	nooutFlag                 = "noout"
 )
 
-func (r *ReconcileClusterDisruption) createPDBForOSD(deployment appsv1.Deployment) error {
+func (r *ReconcileClusterDisruption) createPDB(pdb *policyv1beta1.PodDisruptionBudget) error {
+	err := r.client.Create(context.TODO(), pdb)
+	if err != nil && !kerrors.IsAlreadyExists(err) {
+		return errors.Wrapf(err, "failed to create pdb %q", pdb.Name)
+	}
+	return nil
+}
+
+func (r *ReconcileClusterDisruption) deletePDB(pdb *policyv1beta1.PodDisruptionBudget) error {
+	err := r.client.Delete(context.TODO(), pdb)
+	if err != nil && !kerrors.IsNotFound(err) {
+		return errors.Wrapf(err, "failed to create pdb %q", pdb.Name)
+	}
+	return nil
+}
+
+// createPDBforOSD creates a single PDB for all OSDs with maxUnavailable=1
+// This allows all OSDs in a single failure domain to go down.
+func (r *ReconcileClusterDisruption) createPDBforOSD(namespace string) error {
```

Comment:
Would be helpful to differentiate between which PDB we're creating
```suggestion
func (r *ReconcileClusterDisruption) createOverallPDBforOSD(namespace string) error {
```

Reply:
update the name to `createOverallPDBforOSD`